### PR TITLE
Upgrade to ocamlformat 0.26.0

### PR DESCRIPTION
**Pull request by Julow**
The aim of this commit is to gather feedback.

Changelog can be found here: https://github.com/ocaml-ppx/ocamlformat/blob/main/CHANGES.md

### CODE REVIEW

**Comment by jfeser** [comment]
These look like good changes to me.

**Comment by rgrinberg** [comment]
Yes, no complaints here.

File: ocaml-lsp-server/test/e2e-new/semantic_hl_helpers.ml [hunk 1]
```diff
@@ -52,8 +52,9 @@ let annotate_src_with_tokens ~(legend : SemanticTokensLegend.t)
         "<%s%s-%d>"
         token_types.(!token.type_)
         (if annot_mods then
-         "|" ^ String.concat ~sep:"," (modifiers ~legend:token_mods !token.mods)
-        else "")
+           "|"
+           ^ String.concat ~sep:"," (modifiers ~legend:token_mods !token.mods)
+         else "")
```
Comment (smorimoto) [inline ocaml-lsp-server/test/e2e-new/semantic_hl_helpers.ml:57]: I feel like the old one was better than the new one here.

File: ocaml-lsp-server/src/ocaml_lsp_server.ml [hunk 2]
```diff
@@ -209,8 +209,8 @@ let on_initialize server (ip : InitializeParams.t) =
   let diagnostics =
     Diagnostics.create
       (let open Option.O in
-      let* td = ip.capabilities.textDocument in
-      td.publishDiagnostics)
+       let* td = ip.capabilities.textDocument in
+       td.publishDiagnostics)
```
Comment (smorimoto) [inline ocaml-lsp-server/src/ocaml_lsp_server.ml:213]: Great.

File: ocaml-lsp-server/test/e2e-new/semantic_hl_helpers.ml (hunk 1 above)
Comment (Julow) [reply ocaml-lsp-server/test/e2e-new/semantic_hl_helpers.ml:57]: The `then` branch was 1 indented and aligned with the `if` so I wouldn't say better.

How would you prefer this formatted ?

Comment (Julow) [reply ocaml-lsp-server/test/e2e-new/semantic_hl_helpers.ml:57]: The added indentation causes a line break, which is unfortunate.

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -3,3 +3,5 @@
 # add ocamlformat config `wrap-fun-args=false`
 
 75504946eaa6f817550b649df508d61dde12bbda
+# Upgrade to OCamlformat 0.26.0
+a94c9d608ba4f92a0dce03187e817182084158e0
```
Comment (Julow) [inline .git-blame-ignore-revs:7]: Is this a welcome addition ? If yes, this PR must not be squashed when ready. (it's not ready yet)

**Comment by Julow** [comment]
OCamlformat 0.26.0 has been released! This PR can be merged (not squashed due to `.git-blame-ignore-revs`) or closed if it's a bit too early to upgrade.

**Comment by Julow** [comment]
OCamlformat will be released with compatibility with 5.1 in a few weeks. Let's skip 0.26.0.

**Pull request closed without merging**